frontend/src/lib/components/widgets/DateInput/DateInput.tsx

```diff
@@ -212,6 +212,12 @@ class DateInput extends React.PureComponent<Props, State> {
     const minDate = moment(element.min, DATE_FORMAT).toDate()
     const maxDate = this.getMaxDate()
 
+    // We need to extract the mask and format (date-fns notation) from the provided format string
+    // The user configured date format is based on the momentJS notation and is only allowed to contain
+    // one of YYYY/MM/DD, DD/MM/YYYY, or MM/DD/YYYY" and can also use a period (.) or hyphen (-) as separators.
+    const dateMask = element.format.replaceAll(/[a-zA-Z]/g, "9")
```

Comment:
It's not obvious from the immediate context why we want to replace letters with "9" - can you expand on the comment a bit? (Is it a quirk of UIDatePicker?)

Reply:
That's required by the mask format of the date picker (a 9 means any number):

<img width="281" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/68dec7be-1fdb-4a0b-a056-2f074f64d0dc">

I will add an additional comment.